doc/task_breakdown.md

```diff
@@ -0,0 +1,99 @@
+# SQL Query Builder: Task Breakdown
+
+This is a breakdown of tasks for building the **SQL Query Builder** application.
+
+---
+
+## 1. Interface for Building SQL Queries
+
+Table Selection:
+- A dropdown list to display available tables from the database.
+- The ability to select multiple tables at once.
+- A "Add Table" button to add selected tables to the current query.
+- A "Clear Selection" button to reset selected tables.
+
+Column Selection:
+- Checkbox lists for selecting columns from the chosen tables.
+- Display a message "Please select tables first" if no tables are selected.
+
+Adding Filters (WHERE):
+- Fields to specify conditions:
+- A dropdown list for columns.
+- A dropdown list for operators (`=`, `>`, `<`).
+- A field to input a value.
+- An "Add Filter" button to add new conditions.
+- A "Clear Filters" button to remove all set conditions.
+
+Sorting (ORDER BY):
+- A dropdown list for selecting a column to sort by.
+- The ability to specify the sorting order (ASC or DESC).
+- An "Add Sorting" button to add sorting parameters.
+- A "Clear Sorting" button to reset sorting parameters.
+
+Grouping (GROUP BY):
+- A dropdown list to select columns for grouping results.
+- Display a message "Please select columns for grouping."
+
+Having Clause:
+- Fields to specify conditions on grouped results.
+- A dropdown for columns and operators (`=`, `>`, `<`).
+- A text field to input values for conditions.
+
+---
+
+## 2. Generating and Displaying SQL Queries
```

Comment:
Think about task breakdown as set of completely separated tasks, i.e. you can do one without implementing another